packages/unchained-client/generator/Dockerfile

```diff
@@ -0,0 +1,6 @@
+FROM timbru31/java-node:11-jdk
+
+RUN npm install @openapitools/openapi-generator-cli -g
+RUN openapi-generator-cli version-manager set 5.4.0
```

Comment:
You can combine these two run statements together for a smaller docker layer

Reply:
There will be one less layer yes, but it does not affect the docker image size.

```
[~/github/shapeshift/lib/packages/unchained-client/generator] (main) $ docker images
REPOSITORY                           TAG                                        IMAGE ID       CREATED          SIZE
test2                                latest                                     5e38cf74d467   9 seconds ago    688MB
test1                                latest                                     4f2e1a9de1ac   45 seconds ago   688MB

[~/github/shapeshift/lib/packages/unchained-client/generator] (main) $ docker history 4f2e1a9de1ac
IMAGE          CREATED         CREATED BY                                      SIZE      COMMENT
4f2e1a9de1ac   2 minutes ago   /bin/bash -o pipefail -c #(nop)  ENTRYPOINT …   0B        
9228062fcfba   2 minutes ago   /bin/bash -o pipefail -c openapi-generator-c…   25.3MB    
80a6a5678989   45 hours ago    /bin/bash -o pipefail -c npm install @openap…   32.1MB    
3e054f2ca528   2 days ago      RUN |1 REFRESHED_AT=$(date +%Y-%m-%d) /bin/b…   209MB     buildkit.dockerfile.v0
<missing>      2 days ago      SHELL [/bin/bash -o pipefail -c]                0B        buildkit.dockerfile.v0
<missing>      2 days ago      ENV REFRESHED_AT=$(date +%Y-%m-%d)              0B        buildkit.dockerfile.v0
<missing>      2 days ago      ARG REFRESHED_AT                                0B        buildkit.dockerfile.v0
<missing>      2 days ago      LABEL maintainer=Tim Brust <github@timbrust.…   0B        buildkit.dockerfile.v0
<missing>      19 months ago   /bin/sh -c #(nop)  CMD ["jshell"]               0B        
<missing>      19 months ago   /bin/sh -c #(nop)  ENV JAVA_HOME=/opt/java/o…   0B        
<missing>      19 months ago   /bin/sh -c set -eux;     ARCH="$(dpkg --prin…   322MB     
<missing>      19 months ago   /bin/sh -c #(nop)  ENV JAVA_VERSION=jdk-11.0…   0B        
<missing>      19 months ago   /bin/sh -c apt-get update     && apt-get ins…   35.6MB    
<missing>      19 months ago   /bin/sh -c #(nop)  ENV LANG=en_US.UTF-8 LANG…   0B        
<missing>      19 months ago   /bin/sh -c #(nop)  CMD ["/bin/bash"]            0B        
<missing>      19 months ago   /bin/sh -c mkdir -p /run/systemd && echo 'do…   7B        
<missing>      19 months ago   /bin/sh -c [ -z "$(apt-get indextargets)" ]     0B        
<missing>      19 months ago   /bin/sh -c set -xe   && echo '#!/bin/sh' > /…   745B      
<missing>      19 months ago   /bin/sh -c #(nop) ADD file:4974bb5483c392fb5…   63.2MB    

[~/github/shapeshift/lib/packages/unchained-client/generator] (main) $ docker history 5e38cf74d467
IMAGE          CREATED         CREATED BY                                      SIZE      COMMENT
5e38cf74d467   2 minutes ago   /bin/bash -o pipefail -c #(nop)  ENTRYPOINT …   0B        
d1972350e8b2   2 minutes ago   /bin/bash -o pipefail -c npm install @openap…   57.5MB    
3e054f2ca528   2 days ago      RUN |1 REFRESHED_AT=$(date +%Y-%m-%d) /bin/b…   209MB     buildkit.dockerfile.v0
<missing>      2 days ago      SHELL [/bin/bash -o pipefail -c]                0B        buildkit.dockerfile.v0
<missing>      2 days ago      ENV REFRESHED_AT=$(date +%Y-%m-%d)              0B        buildkit.dockerfile.v0
<missing>      2 days ago      ARG REFRESHED_AT                                0B        buildkit.dockerfile.v0
<missing>      2 days ago      LABEL maintainer=Tim Brust <github@timbrust.…   0B        buildkit.dockerfile.v0
<missing>      19 months ago   /bin/sh -c #(nop)  CMD ["jshell"]               0B        
<missing>      19 months ago   /bin/sh -c #(nop)  ENV JAVA_HOME=/opt/java/o…   0B        
<missing>      19 months ago   /bin/sh -c set -eux;     ARCH="$(dpkg --prin…   322MB     
<missing>      19 months ago   /bin/sh -c #(nop)  ENV JAVA_VERSION=jdk-11.0…   0B        
<missing>      19 months ago   /bin/sh -c apt-get update     && apt-get ins…   35.6MB    
<missing>      19 months ago   /bin/sh -c #(nop)  ENV LANG=en_US.UTF-8 LANG…   0B        
<missing>      19 months ago   /bin/sh -c #(nop)  CMD ["/bin/bash"]            0B        
<missing>      19 months ago   /bin/sh -c mkdir -p /run/systemd && echo 'do…   7B        
<missing>      19 months ago   /bin/sh -c [ -z "$(apt-get indextargets)" ]     0B        
<missing>      19 months ago   /bin/sh -c set -xe   && echo '#!/bin/sh' > /…   745B      
<missing>      19 months ago   /bin/sh -c #(nop) ADD file:4974bb5483c392fb5…   63.2MB 
```

Reply:
@kaladinlight [Minimizing the number of docker layer's](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#minimize-the-number-of-layers) is primarily about the number of layers created for the image and not necessarily the final docker image size.